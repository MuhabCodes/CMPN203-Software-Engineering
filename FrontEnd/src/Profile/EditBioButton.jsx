/* eslint-disable react/prop-types */
import React from 'react';
import edit from './assets/edit_icon.png';
import './AboutBio.css';
// eslint-disable-next-line react/prop-types
/**
 * This component renders a button that allows the user to show or hide textarea to edit bio
 * @component EditBioButton
 * @function EditBioButton
 * @example <EditBioButton/>
 * @param {string} text - description text of user
 * @returns Edit bio button
 */
const EditBioButton = ({ text }) => (
  <button
    data-testid="edit-bio"
    aria-label="editingbutton"
    type="button"
    id="user-edit-biobtn"
    className="edit-button"
    onClick={() => { // hide text area and show edit text area
      document.getElementById('textcontainer').style.display = 'none';
      // eslint-disable-next-line react/prop-types
      if (text.length > 200) {
        document.getElementById('readbtn').style.display = 'none';
      }
      document.getElementById('editingarea').style.display = 'block';
      document.getElementById('usertextarea').innerHTML = text;
    }}
  >
    <img src={edit} alt="" className="edit-button-img" />
  </button>
);
export default EditBioButton;
