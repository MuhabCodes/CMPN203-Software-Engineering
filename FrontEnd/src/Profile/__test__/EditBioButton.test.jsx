import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import EditBioButton from '../EditBioButton';
import '@testing-library/jest-dom/extend-expect';

it('it renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<EditBioButton />, div);
});

it('renders the button correctly', () => {
  const { getByTestId } = render(<EditBioButton />);
  expect(getByTestId('edit-bio')).toBeTruthy();
});

it('matches a snapshot', () => {
  const tree = renderer.create(<EditBioButton />).toJSON();
  expect(tree).toMatchSnapshot();
});
