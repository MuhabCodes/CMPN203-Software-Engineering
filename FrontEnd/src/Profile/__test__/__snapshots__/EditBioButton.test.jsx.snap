// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches a snapshot 1`] = `
<button
  aria-label="editingbutton"
  className="edit-button"
  data-testid="edit-bio"
  id="user-edit-biobtn"
  onClick={[Function]}
  type="button"
>
  <img
    alt=""
    className="edit-button-img"
    src="edit_icon.png"
  />
</button>
`;
